RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -95,39 +90,49 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const ChamberHitContainer& u
   BoolContainer used(rechits.size(), false);
   BoolContainer used3p(rechits.size(), false);
   // This is going to point to fits to hits, and its content will be used to create a CSCSegment
-  sfit_ = 0;
+  AlgoState aState;
+  aState.aChamber = aChamber;
+  aState.doCollisions = doCollisions;
+  aState.dRMax = dRMax;
+  aState.dPhiMax = dPhiMax;
+  aState.dRIntMax = dRIntMax;
+  aState.dPhiIntMax = dPhiIntMax;
+  aState.chi2Norm_2D_ = chi2Norm_2D_;
+  aState.chi2_str_ = chi2_str_;
+  aState.chi2Max = chi2Max;
+
   // Define buffer for segments we build
   std::vector<CSCSegment> segments;
   ChamberHitContainerCIt ib = rechits.begin();
   ChamberHitContainerCIt ie = rechits.end();
   // Possibly allow 3 passes, second widening scale factor for cuts, third for segments from displaced vertices
-  windowScale = 1.; // scale factor for cuts
+  aState.windowScale = 1.; // scale factor for cuts
   bool search_disp = false;
-  strip_iadd = 1;
-  chi2D_iadd = 1;
+  aState.strip_iadd = 1;
+  aState.chi2D_iadd = 1;
   int npass = (wideSeg > 1.)? 3 : 2;
   for (int ipass = 0; ipass < npass; ++ipass) {
-    if(windowScale >1.){
+    if(aState.windowScale >1.){
       iadd = 1;
-      strip_iadd = 2;
-      chi2D_iadd = 2;
+      aState.strip_iadd = 2;
+      aState.chi2D_iadd = 2;
     }
     int used_rh = 0;
     for (ChamberHitContainerCIt i1 = ib; i1 != ie; ++i1) {
       if(used[i1-ib])used_rh++;
     }
 
     //change the tresholds if it's time to look for displaced mu segments
-    if(doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
-      doCollisions = false;
-      windowScale = 1.; // scale factor for cuts
-      dRMax = 2.0;
-      dPhiMax = 2*dPhiMax;
-      dRIntMax = 2*dRIntMax;
-      dPhiIntMax = 2*dPhiIntMax;
-      chi2Norm_2D_ = 5*chi2Norm_2D_;
-      chi2_str_ = 100;
-      chi2Max = 2*chi2Max;
+    if(aState.doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
```

Comment:
Could you please also apply here the (unrelated) fix pointed out in https://github.com/cms-sw/cmssw/pull/19081#pullrequestreview-43183950?